packages/store/addon/-private/system/relationships/state/relationship.ts

```diff
@@ -227,7 +227,14 @@ export default class Relationship {
         return;
       }
       let relationship = relationshipStateFor(inverseRecordData, this.inverseKey);
-      relationship.inverseDidDematerialize(this.recordData);
+      let inverseRecordDataRelationship = inverseRecordData.getBelongsTo(this.inverseKey!)
+        ._relationship!.inverseRecordData;
```

Comment:
What are the ! Characters here for?

Reply:
The non-null assertion on `this.inverseKey` was because the TypeScript compiler hasn't propagated the null check on it from the start of the function. I've refactored the code slightly to avoid the need for the `!` here.

Have also added a check that the `_relationship` member here is actually set, in place of the non-null assertion.

Reply:
Thanks!